contract/pkg/messages.go

```diff
@@ -54,3 +55,18 @@ type BucketStatus struct {
 func (s *BucketStatus) RentExpired() bool {
 	return s.RentCoveredUntilMs < uint64(time.Now().UnixMilli())
 }
+
+func (s *BucketStatus) HasWriteAccess(publicKey string) (bool, error) {
+	address, err := types.NewAddressFromHexAccountID(publicKey)
```

Comment:
With same seed/secretPhrase system generates different public keys for each sign algorithm. Did you test it or now we will be able to test only sr25519 sign algorithm?

Reply:
in unit test below i've used ed25519 keys so it suppose to work with different algorithms. The one requirement here is that writerId should have the same algorithm. At the moment in writerId only owner is added, and polkadot extension uses sr25519, so the same algo should be used for you pub/priv key pair you are going to use in library.

Once there is a grantBucketPermission method that can accept accountId, ed25519 can be supported as well. You just need to pass accountId based on ed25519 public key to the grantBucketPermission method